controller/control.go

```diff
@@ -1035,3 +1006,68 @@ func (c *Controller) monitoring(address string, backend types.Backend) {
 	logrus.Infof("Monitoring stopped %v", address)
 	c.RemoveReplicaNoLock(address)
 }
+
+// monitorPreload This function should only be called for the first replica
+func (c *Controller) monitorPreload(address string) {
+	var (
+		sendSignal bool
+		status     string
+		err        error
+	)
+	logrus.Infof("Get preload status of backend %s", address)
+	verifyPreload := true
+	for verifyPreload {
+		c.Lock()
+		if status, err = c.backend.GetPreloadStatus(address); err != nil {
+			logrus.Errorf("Failed to monitor preload, error: %v", err)
+			c.Unlock()
+			return
+		}
+		c.Unlock()
+		switch types.PreloadStatus(status) {
+		case types.Done:
+			verifyPreload = false
+			logrus.Infof("Preload is completed on backend %s", address)
+		case types.Started:
+			logrus.Warningf("Preload is not done yet on backend %s, current status: %s", address, status)
+			time.Sleep(2 * time.Second)
+		case types.Error:
+			logrus.Errorf("Preload status of backend is %s", status)
+			return
+		default:
+			logrus.Warningf("Preload is not done yet on backend %s, current status: %s", address, status)
+			time.Sleep(2 * time.Second)
+		}
+	}
+	c.Lock()
+	c.startFrontend()
+	c.setReplicaModeNoLock(address, types.RW)
+	c.UpdateVolStatus()
+	for regrep := range c.RegisteredReplicas {
+		sendSignal = true
+		for _, tmprep := range c.replicas {
+			if tmprep.Address == "tcp://"+regrep+":9502" {
+				sendSignal = false
+				break
+			}
+		}
+		if sendSignal {
+			logrus.Infof("sending add signal to %v", regrep)
+			c.factory.SignalToAdd(regrep, "add")
```

Comment:
This signaling after preload is important. In previous versions, this was missed, and would have caused some data issue.